plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -184,15 +244,30 @@ private void processPredictions(
             }
         }
 
-        ActionRequest request = getRemoteModelInferenceRequest(modelParameters, inferenceProcessorAttributes.getModelId());
+        Set<String> inputMapKeys = new HashSet<>(modelParameters.keySet());
+        inputMapKeys.removeAll(modelConfigs.keySet());
```

Comment:
I don't get it about this logic. Looks like `modelParameter` is same as `modelConfigs` which both are initialized by `inferenceProcessorAttributes.getModelConfigMaps()`.

Reply:
`modelParameters` basically consists of all the input fields and values that are sent as part of model inference. It is updated [here](https://github.com/rbhavna/ml-commons/blob/ml_ingest_processor/plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java#L357) to include the input fields from ingest document along with model_configs defined in the processor.

What we are doing from line 247 to 252 is separating the inputs that come from ingest document such as `text_docs` and adding them to different map. 

This is because in remote model, we send all inputs and config fields as part of one map called `parameters` But local models need them to be sent as separate fields